feat: Add reusable deploy workflow

losvedir opened the pull request:
Allows another app to reuse it, e.g.:

**my_app/.github/workflows/deploy.yml**
```   
name: Deploy to ECS

on:
  workflow_dispatch:
    inputs:
      environment:
        type: environment
        required: true
        default: dev
  push:
    branches: master

jobs:
  call-workflow:
    uses: mbta/actions/workflows/deploy@main
    with:
      app-name: my-app
      environment: ${{ github.event.inputs.environment }}
    secrets:
      aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
      aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
      docker-repo: ${{ secrets.DOCKER_REPO }}
      slack-webhook: ${{ secrets.SLACK_WEBHOOK }}
```

losvedir committed:
feat: Add reusable deploy workflow

paulswartz reviewed (commented):
Could you add your example in the commit description as a `README.md` in this folder? Otherwise it might be hard for folks to figure out how to use this in their own workflows.

losvedir commented:
Pushed up a revised commit. I added a `README.md` to the directory. I also renamed the workflow to `deploy-ecs`, to be consistent with the workflow template in the other repo (and since potentially we'll have other deploy types, like to Lambda).
